layouts/partials/head.html

```diff
@@ -52,3 +54,18 @@
     {{ template "_internal/google_analytics_async.html" . -}}
   {{ end -}}
 {{ end -}}
+
+{{ define "algolia/head" -}}
+
+{{ if and .Site.Params.search (isset .Site.Params.search "algolia") -}}
```

Comment:
Since minimum required version is hugo v0.110 now, what about fetching the docsearch npm resource at build time rather than at runtime? This way, we also have better GDPR support (#1354). 
Also, I amended the code so that users now can specify the wanted docsearch version inside thier project's `hugo.yaml`:

```
params:
  search:
    algolia:
      version: 3.5.1
```

```suggestion
{{ if and .Site.Params.search (isset .Site.Params.search "algolia") -}}
  {{ $version := "v3.5.1" -}}
  {{ with .Site.Params.search.algolia.version -}}
    {{ $version = . -}}
  {{ end -}}
  {{ $cdnurl := printf "https://cdn.jsdelivr.net/npm/@docsearch/css@%s" $version -}}
  {{ if eq (resources.GetRemote $cdnurl) nil -}}
    {{ errorf "Invalid Algolia version %s, could not retrieve this version from CDN" $version -}}
  {{ end -}}

  {{ with resources.GetRemote $cdnurl -}}
    {{ with .Err -}}
      {{ errorf "Error while retrieving resource %q from CDN. Status: %s." $cdnurl .Data.Status -}}
    {{ else -}}
      {{ $secureCSS := . | resources.Fingerprint "sha512" -}}
<link rel="stylesheet" href="{{ .RelPermalink }}" integrity="{{ $secureCSS.Data.Integrity }}" crossorigin="anonymous" />
    {{ end -}}
  {{ end -}}
{{ end -}}

```